src/commands/soundbot/IgnoreCommand.ts

```diff
@@ -0,0 +1,22 @@
+import { Permissions } from 'discord.js';
+
+import IgnoreBaseCommand from '../base/IgnoreBaseCommand';
+
+export class IgnoreCommand extends IgnoreBaseCommand {
+  public readonly USAGE = 'Usage: !ignore <user>';
+
+  public run() {
+    // @REVIEW How to extract this method into a Interface / Mixin, so that
+    // all restricted Commands can have call the same method instead of this duplication
+    // Another subclass could be potentially bad.
+    if (!this.message.member.hasPermission(Permissions.FLAGS.ADMINISTRATOR!)) return;
```

Comment:
I will have to think about this. Maybe you can use Typescript Decorators for this?